components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java

```diff
@@ -238,22 +239,39 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
         response.sendRedirect(getRedirectURL(redirectURL, request));
     }
 
+    private String getOPBSCookieValue(HttpServletRequest request) {
+
+        Cookie opBrowserStateCookie = OIDCSessionManagementUtil.getOPBrowserStateCookie(request);
+        if (opBrowserStateCookie != null) {
+            return opBrowserStateCookie.getValue();
+        } else {
+            String sessionDataKey = request.getParameter(OIDCSessionConstants.OIDC_SESSION_DATA_KEY_PARAM);
+            if (StringUtils.isNotBlank(sessionDataKey)) {
+                OIDCSessionDataCacheEntry cacheEntry = getSessionDataFromCache(sessionDataKey);
+                if (cacheEntry != null) {
+                    return cacheEntry.getParamMap().get(OIDCSessionConstants.OPBS_COOKIE_ID);
+                }
+            }
+        }
+        return null;
+    }
+
     /**
      * If postLogoutRedirectUri is send in Logout request parameter then set it as redirect URL.
      *
      * @param redirectURL
-     * @param opBrowserStateCookie
+     * @param opBrowserStateCookieValue
```

Comment:
Can we add a param description

Reply:
Addressed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1519/commits/b6b16b4e9375169a8e8109dead588f7651e3a54b.